pkgs/applications/audio/musescore/default.nix

```diff
@@ -140,6 +140,26 @@ in stdenv'.mkDerivation (finalAttrs: {
     ln -s $out/Applications/mscore.app/Contents/MacOS/mscore $out/bin/mscore
   '';
 
+  # muse-sounds-manager installs Muse Sounds sampler libMuseSamplerCoreLib.so.
+  # It requires that argv0 of the calling process ends with "/mscore" or "/MuseScore-4".
+  # We need to ensure this in two cases:
+  #
+  # 1) when the user invokes MuseScore as "mscore" on the command line or from
+  #    the .desktop file, and the normal argv0 is "mscore" (no "/");
+  # 2) when MuseScore invokes itself via File -> New, and the normal argv0 is
+  #    the target of /self/proc/exe, which in Nixpkgs was "{...}/.mscore-wrapped"
+  #
+  # In order to achieve (2) we install the final binary as $out/mscore, and
+  # in order to achieve (1) we use makeWrapper without --inherit-argv0.
+  #
+  # wrapQtAppsHook uses wrapQtApp -> wrapProgram -> makeBinaryWrapper --inherit-argv0
+  # so we disable it and explicitly use makeQtWrapper.
+  dontWrapQtApps = stdenv.isLinux;
+  postFixup = lib.optionalString stdenv.isLinux ''
+    mv $out/bin/mscore $out
+    makeQtWrapper $out/mscore $out/bin/mscore
+  '';
```

Comment:
It feels a bit odd to do that. If I understand correctly, using `makeQtWrapper` with `--argv mscore` should be equivalent?

Reply:
> It feels a bit odd to do that. If I understand correctly, using `makeQtWrapper` with `--argv mscore` should be equivalent?

Oh wait never mind, I understand now - we can't wrap `.mscore-wrapped`, that's the main point.

Reply:
I'd still like to put that wrapped `mscore` somewhere else - how about `$out/libexec/mscore`?

Reply:
`makeQtWrapper --argv0 $out/mscore` (we need argv0 to end with "/mscore") would be equivalent, but then `--argv0` is unnecessary because this is how executing programs by absolute path normally works: their argv0 is that absolute path.

Moving `$out/bin/mscore` out of `$out/bin` is necessary because both the end-user wrapper and the wrapped binary must be named `mscore`, and the wrapper should be left in `$out/bin/mscore`.

Reply:
I can move it to libexec.

Reply:
> Moving `$out/bin/mscore` out of `$out/bin` is necessary because both the end-user wrapper and the wrapped binary must be named `mscore`, and the wrapper should be left in `$out/bin/mscore`.

I understand that, but you don't state that there is a problem with using `$out/libexec/mscore` as the path to the wrapped executable right?

Reply:
> I can move it to libexec.

Wait maybe don't do that, due to this:

https://github.com/NixOS/nixpkgs/blob/31437c831582f3b7c54db408fbcbd3e9ae264648/pkgs/development/libraries/qt-5/hooks/wrap-qt-apps-hook.sh#L79

Let's go for `$out/lib/mscore`

Reply:
No, I did not think of it myself but libexec is exactly the right place, thank you for the suggestion!
wrap-qt-apps-hook touches it exactly because it is a place for executables, but we are not concerned with that because we use dontWrapQtApps.

Reply:
> No, I did not think of it myself but libexec is exactly the right place, thank you for the suggestion! wrap-qt-apps-hook touches it exactly because it is a place for executables, but we are not concerned with that because we use dontWrapQtApps.

Right, I was just thinking about it :).